integrated_channels/canvas/client.py

```diff
@@ -51,9 +51,123 @@ def __init__(self, enterprise_configuration):
+            # anything else will prevent other transmissions to happen.
+            raise CanvasClientError(
+                "No Canvas user ID found associated with email: {}".format(learner_data['userID'])
+            )
+
+        # With the Canvas user ID, retrieve all courses for the user.
+        get_users_courses_url = '{canvas_base_url}/api/v1/users/{canvas_user_id}/courses'.format(
+            canvas_base_url=self.enterprise_configuration.canvas_base_url,
+            canvas_user_id=canvas_user_id
+        )
+        rsps = self.session.get(get_users_courses_url)
+        rsps.raise_for_status()
+        user_courses_response = rsps.json()
+
+        # Find the course who's integration ID matches the learner data course ID
+        # Is there a better identifier here?
+        course_id = None
+        for course in user_courses_response:
+            integration_id = course['integration_id']
+            if '+'.join(integration_id.split(":")[1].split("+")[:2]) == learner_data['courseID']:
+                course_id = course['id']
+                break
+
+        if not course_id:
+            raise CanvasClientError(
+                "Course: {course_id} not found registered in Canvas for Edx learner: {user_id}"
+                "/Canvas learner: {canvas_user_id}.".format(
+                    course_id=learner_data['courseID'],
+                    user_id=learner_data['userID'],
+                    canvas_user_id=canvas_user_id
+                ))
+
+        # Check if the course assignment already exists
+        canvas_assignments_url = '{canvas_base_url}/api/v1/courses/{course_id}/assignments'.format(
+            canvas_base_url=self.enterprise_configuration.canvas_base_url,
+            course_id=course_id
+        )
+        resp = self.session.get(canvas_assignments_url)
+        resp.raise_for_status()
+        assignments_resp = resp.json()
+        assignment_id = None
+        for assignment in assignments_resp:
+            try:
+                if assignment['integration_id'] == integration_id:
+                    assignment_id = assignment['id']
+                    break
+            except (KeyError, ValueError):
+                # This catches if the response the client got back doesn't have the expected assignment
+                # data.
+                raise CanvasClientError(
+                    "Something went wrong retrieving assignments from Canvas. Got response: {}".format(resp.text)
+                )
+
+        # Canvas requires a course assignment for a learner to be assigned a grade.
+        # If no assignment has been made yet, create it.
+        if not assignment_id:
+            assignment_creation_data = {
+                'assignment': {
+                    'name': 'Final Grade',
+                    'submission_types': 'none',
+                    'integration_id': integration_id,
+                    'published': True,
+                    'points_possible': 100
```

Comment:
Yup, I can dictate what type of grade is being received here, but since I saw that grade percentage was being exported by the learning api, I figured that was the most clear first iteration on it. We can possibly dig into what type of grading the class had?

Reply:
Small note- I also believe that the type of grade has to be dictated on the assignment level.

Reply:
Hmm ok, looking at the GradesAPI, it seems like we're retrieving a letter grade and a percent grade. In my mind the percent grade was more accurate data to report as completion data to Canvas.

Reply:
ah I see, as long as we will stay consistent here and send correct value for points_possible and the grade, I am good! just wanted to check on that